apps/volunteer/sign_up.py

```diff
@@ -136,3 +130,56 @@ def account():
     form.allow_comms.data = volunteer.allow_comms_during_event
 
     return render_template("volunteer/account.html", user=current_user, form=form)
+
+
+def generate_arrival_options():
+    choices = []
+
+    # Work out our first arrival based on config
+    first_arrival = event_start() - timedelta(app.config["ARRIVAL_DAYS"])
+
+    # Add the first date to the list with or earlier
+    choices.append((
+        first_arrival.strftime("%F"),
+        first_arrival.strftime("%A %-d %B or earlier")
+    ))
+
+    # Work out dates between first arrival and end of the event
+    days = period(
+        first_arrival + timedelta(days=1),
+        event_end()).range("days", 1)
+
+    # Add each date to the list
+    for d in days:
+        choices.append((
+            d.strftime("%F"),
+            d.strftime("%A %-d %B")
+        ))
```

Comment:
There may be a few changes we can make to reduce the line-count of the `generate_arrival_options` and `generate_departure_options` methods.

It looks like there's near-duplication of code as a result of handling the '<date> or earlier' and '<date> or later' special case suffixes.

Here's something I've been experimenting with locally which reduces some of the date formatting duplication:

```
    date = event_start() - timedelta(app.config["ARRIVAL_DAYS"])

    choices.append(render_date_option(date, suffix=" or earlier"))  # calls a shared method
    while date < event_end() - timedelta(days=1):
        date += timedelta(days=1)
        choices.append(render_date_option(date))
```

What do you reckon?

Reply:
Possibly, though I'm personally not a fan of reducing line length if it also reduces readability.

This is long form and can probably be refactored to reduce the duplication, but equally I'd like to move on to other issues now this is fundamentally working.

I think this is good enough for now, though opinions may vary ;) If the consensus is that it's not good enough then I'll come back to it again.

Reply:
Re-instated the `generate_day_options` function and made us of it.

I am unconvinced that it adds anything but consensus seems to be that we want it so here it is.